util/parser/expression.go

```diff
@@ -36,8 +70,18 @@ func ParseExpression(expr string) (node ast.ExprNode, err error) {
 	charset, collation := getDefaultCharsetAndCollate()
 	stmts, err := parser.New().Parse(expr, charset, collation)
 	if err == nil {
-		sel := stmts[0].(*ast.SelectStmt)
-		node = sel.Fields.Fields[0].Expr
+		node = stmts[0].(*ast.SelectStmt).Fields.Fields[0].Expr
+	}
+	return node, errors.Trace(err)
+}
+
+// SimpleResolveName resolves all column names in the expression node.
+func SimpleResolveName(node ast.ExprNode, tblInfo *model.TableInfo) (ast.ExprNode, error) {
+	nr := nameResolver{tblInfo, nil}
+	if _, ok := node.Accept(&nr); !ok {
+		return nil, errors.Trace(nr.err)
 	}
+	sc := new(variable.StatementContext) // here we use the default StatementContext.
```

Comment:
I'm afraid using default StatementContext may sometime panics, when `sc.TimeZone` is nil and `expression.InferType` is handling timestamp datum.